src/main/java/com/checkout/forex/four/QuoteResponse.java

```diff
@@ -0,0 +1,34 @@
+package com.checkout.forex.four;
+
+import com.checkout.common.Currency;
+import com.google.gson.annotations.SerializedName;
+import lombok.Data;
+
+import java.time.Instant;
+
+@Data
+public final class QuoteResponse {
+
+    private String id;
+
+    @SerializedName("source_currency")
+    private Currency sourceCurrency;
+
+    @SerializedName("source_amount")
+    private Long sourceAmount;
+
+    @SerializedName("destination_currency")
+    private Currency destinationCurrency;
+
+    @SerializedName("destination_amount")
+    private Long destinationAmount;
+
+    private Number rate;
```

Comment:
What about making it a Double?

Reply:
I thought about that too, but in the project I found that we are using more a Number object

Reply:
I was only able to find an usage on `Number` in the dispute module and actually it seems to be wrong. Number is an abstract class and apparently Gson does us the favor of instantiating a `LazilyParserNumber`. Please change to `Double` and do the same on `Dispute` and `DisputeDetailsResponse`.